File: windows/CodexFlowWindows/src/CodexFlowWindows/Data/Models/EventModels.cs
using System.Text.Json.Serialization;

namespace CodexFlowWindows.Data.Models;

public sealed record AgentEvent
{
    [JsonPropertyName("type")]
    public string Type { get; init; } = string.Empty;

    [JsonPropertyName("timestamp")]
    public string Timestamp { get; init; } = string.Empty;

    [JsonPropertyName("payload")]
    public JsonValue Payload { get; init; } = JsonValue.Null;
}

File: android/CodexFlowAndroid/app/src/main/java/com/codexflow/codexflow/data/model/EventModels.kt
package com.codexflow.codexflow.data.model

import kotlinx.serialization.Serializable

@Serializable
data class AgentEvent(
    val type: String = "",
    val timestamp: String = "",
    val payload: JsonValue = JsonValue.NullValue
)

enum class SseConnectionState {
    Disconnected,
    Connecting,
    Connected,
    Reconnecting
}
